### PR TITLE
Release/10.1.4

### PR DESCRIPTION
**Pull request by JNdhlovu**
Story: https://app.shortcut.com/smileid/story/12503/add-offline-mode-functionality-to-v10-wrappers-react-native

## Summary
* Support for react native 0.74.x
* Android minSdkVersion 23 as per react native 0.74.x (https://reactnative.dev/blog/2024/04/22/release-0.74#android-minimum-sdk-bump-android-60)
* Added offline functionality
  * setAllowOfflineMode
  * submitJob
  * getUnsubmittedJobs
  * getSubmittedJobs
  * cleanup
* Bump android to 10.1.6 (https://github.com/smileidentity/android/releases/tag/v10.1.6)
* Bump iOS to 10.2.0 (https://github.com/smileidentity/ios/releases/tag/v10.2.0)

### CODE REVIEW

**Review by jumaallan — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,22 +1,18 @@
...
-
+name: CI
 on:
-  workflow_call:
   push:
     branches:
       - main
   pull_request:
-
-concurrency:
```
Comment (vanshg) [inline .github/workflows/ci.yml:10]: Why remove the concurrency group?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,22 +1,18 @@
...
+    branches:
+      - main
 
 jobs:
   lint:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
-        uses: actions/checkout@v4
+        uses: actions/checkout@v3
```
Comment (vanshg) [inline .github/workflows/ci.yml:15]: Why are we downgrading versions?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -111,27 +109,22 @@ jobs:
...
-        with:
-          xcode-version: 15.2
+        uses: actions/checkout@v3
 
       - name: Setup
         uses: ./.github/actions/setup
 
       - name: Cache turborepo for iOS
-        uses: actions/cache@v4
+        uses: actions/cache@v3
```
Comment (vanshg) [inline .github/workflows/ci.yml:118]: same here

File: .yarn/releases/yarn-3.6.1.cjs
Comment (vanshg) [inline .yarn/releases/yarn-3.6.1.cjs:1]: Is this the entire yarn tool checked in to the repo?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,17 +1,16 @@
 # Release Notes
 
-## 10.1.5
-
-* Bump iOS to 10.2.0
-
 ## 10.1.4
 
+* Support for react native 0.74.x
```
Comment (vanshg) [inline CHANGELOG.md:5]: Are there any migration steps that will be required due to the change in architecture?

**Review by vanshg — CHANGES_REQUESTED** [review]

**Review by vanshg — APPROVED** [review]

**Pull request merged**